dd-java-agent/instrumentation/play-2.8/src/main/java8/datadog/trace/instrumentation/play28/PlayHttpServerDecorator.java

```diff
@@ -0,0 +1,96 @@
+// Modified by SignalFx
+package datadog.trace.instrumentation.play28;
+
+import datadog.trace.agent.decorator.HttpServerDecorator;
+import datadog.trace.api.DDTags;
+import datadog.trace.instrumentation.api.AgentSpan;
+import datadog.trace.instrumentation.api.Tags;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.UndeclaredThrowableException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import lombok.extern.slf4j.Slf4j;
+import play.api.mvc.Request;
+import play.api.mvc.Result;
+import play.api.routing.HandlerDef;
+import play.routing.Router;
+import scala.Option;
+
+
+@Slf4j
+public class PlayHttpServerDecorator extends HttpServerDecorator<Request, Request, Result> {
+  public static final PlayHttpServerDecorator DECORATE = new PlayHttpServerDecorator();
+
+  @Override
+  protected String[] instrumentationNames() {
+    return new String[] {"play"};
+  }
+
+  @Override
+  protected String component() {
+    return "play-action";
+  }
+
+  @Override
+  protected String method(final Request httpRequest) {
+    return httpRequest.method();
+  }
+
+  @Override
+  protected URI url(final Request request) throws URISyntaxException {
+    return new URI((request.secure() ? "https://" : "http://") + request.host() + request.uri());
+  }
+
+  @Override
+  protected String peerHostname(final Request request) {
+    return null;
+  }
+
+  @Override
+  protected String peerHostIP(final Request request) {
+    return request.remoteAddress();
+  }
+
+  @Override
+  protected Integer peerPort(final Request request) {
+    return null;
+  }
+
+  @Override
+  protected Integer status(final Result httpResponse) {
+    return httpResponse.header().status();
+  }
+
+  @Override
+  public AgentSpan onRequest(final AgentSpan span, final Request request) {
+    super.onRequest(span, request);
+    if (request != null) {
+      // more about routes here:
+      // https://github.com/playframework/playframework/blob/master/documentation/manual/releases/release26/migration26/Migration26.md
+      final Option<HandlerDef> defOption =
+        request.attrs().get(Router.Attrs.HANDLER_DEF.asScala());
+      if (!defOption.isEmpty()) {
+        final String path = defOption.get().path();
+        span.setTag(DDTags.RESOURCE_NAME, path);
+      }
+    }
+    return span;
+  }
+
+  @Override
+  public AgentSpan onError(final AgentSpan span, Throwable throwable) {
+    span.setTag(Tags.HTTP_STATUS, 500);
```

Comment:
It's advisable to only set the status if made available by a response object.  I'd imagine it's possible for this to be incorrect and could cause issues for users.